File: src/main/java/com/tiobe/plugins/intellij/actions/AnalyzeProject.java
package com.tiobe.plugins.intellij.actions;

import com.intellij.execution.configurations.GeneralCommandLine;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;

public class AnalyzeProject extends AbstractAnalyzeAction {
    public GeneralCommandLine getTICSCommand(final VirtualFile file, final Project project) {
        final GeneralCommandLine cmd = new GeneralCommandLine(TICS_COMMAND);
        cmd.addParameters(IDE_PARAMETER, IDE_NAME);
        final VirtualFile projectFile = project == null ? null : project.getProjectFile();
        if (projectFile != null) {
            cmd.addParameters(PROJFILE_PARAMETER, projectFile.getCanonicalPath());
        }
        final VirtualFile baseDir = project == null ? null : project.getBaseDir();
        if (baseDir != null) {
            cmd.addParameters(baseDir.getCanonicalPath());
        }
        return cmd;
    }

    @Override
    public boolean isFileRequired() {
        return false;
    }

    @Override
    public boolean isProjectRequired() {
        return true;
    }
}

File: src/main/kotlin/com/tiobe/plugins/intellij/actions/AnalyzeProject.kt
package com.tiobe.plugins.intellij.actions

import com.intellij.execution.configurations.GeneralCommandLine
import com.intellij.openapi.project.Project
import com.intellij.openapi.project.guessProjectDir
import com.intellij.openapi.vfs.VirtualFile

class AnalyzeProject : AnalyzeAction() {
    override fun getTicsCommand(file: VirtualFile?, project: Project?): GeneralCommandLine {
        val command = GeneralCommandLine(TICS_COMMAND)
        command.addParameters(IDE_PARAMETER, IDE_NAME)

        project?.projectFile?.canonicalPath?.let {
            command.addParameters(PROJECT_FILE_PARAMETER, it)
        }

        project?.guessProjectDir()?.canonicalPath?.let {
            command.addParameter(it)
        }

        return command
    }

    override fun isFileRequired(): Boolean {
        return false
    }

    override fun analyzeFileOpenedInEditor(): Boolean {
        return false
    }
}